internal/providers/pluginfw/resources/sharing/resource_share.go

```diff
@@ -0,0 +1,395 @@
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "created_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "created_by")
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "object", "added_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "object", "added_by")
+
+		c.SetRequired("object", "data_object_type")
+		c.SetRequired("object", "partitions", "values", "op")
+		c.SetRequired("object", "partitions", "values", "name")
+		return c
+	})
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Share",
+		Attributes:  attrs,
+		Blocks:      blocks,
+	}
+}
+
+func (d *ShareResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if d.Client == nil && req.ProviderData != nil {
+		d.Client = pluginfwcommon.ConfigureResource(req, resp)
+	}
+}
+
+func (r *ShareResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	var plan ShareInfoExtended
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var planGoSDK sharing.ShareInfo
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, plan, &planGoSDK)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createShare sharing.CreateShare
+	resp.Diagnostics.Append(converters.TfSdkToGoSdkStruct(ctx, plan, &createShare)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	shareInfo, err := w.Shares.Create(ctx, createShare)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create share", err.Error())
+		return
+	}
+
+	shareChanges := shareChanges(planGoSDK, string(sharing.SharedDataObjectUpdateActionAdd))
+
+	updatedShareInfo, err := w.Shares.Update(ctx, shareChanges)
+	if err != nil {
+		// delete orphaned share if update fails
+		if d_err := w.Shares.DeleteByName(ctx, shareInfo.Name); d_err != nil {
+			resp.Diagnostics.AddError("failed to delete orphaned share", d_err.Error())
+			return
+		}
+		resp.Diagnostics.AddError("failed to update share", err.Error())
+		return
+	}
+
+	matchOrder(updatedShareInfo.Objects, planGoSDK.Objects, func(obj sharing.SharedDataObject) string { return obj.Name })
+
+	var newState ShareInfoExtended
+	resp.Diagnostics.Append(converters.GoSdkToTfSdkStruct(ctx, updatedShareInfo, &newState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	newState.SyncEffectiveFieldsDuringCreateOrUpdate(plan.ShareInfo)
+	for i := range newState.Objects {
+		newState.Objects[i].SyncEffectiveFieldsDuringCreateOrUpdate(plan.Objects[i])
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, newState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *ShareResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	logger.SetTfLogger(logger.NewTfLogger(ctx))
```

Comment:
I think we should do this while we configure the client so it gets set for each CRUD in all resources like we do in sdkv2 :https://github.com/databricks/terraform-provider-databricks/blob/main/internal/providers/sdkv2/sdkv2.go#L235. 

It doesn't have to be in this PR so we can remove this and create a ticket to track adding this properly.

Reply:
Heh, no this is just sloppy leftovers from debugging. Oops!